streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -108,16 +104,31 @@ public long size() {
     }
 
     @Override
-    public final boolean equals(Object o) {
+    public Windows<Window> until(final long duration) throws IllegalArgumentException {
```

Comment:
Change return type to `JoinWindows` (same for all other classed inheriting from `Windows`)